pick.c

```diff
@@ -649,12 +608,72 @@ tty_init(void)
 
 	setupterm((char *)0, fileno(tty_out), (int *)0);
 
-	choices_lines = lines - 1;	/* available lines, minus query line */
+	tty_size();
 
 	if (use_alternate_screen)
 		tty_putp(enter_ca_mode, 0);
 
 	signal(SIGINT, handle_sigint);
+#if HAVE_DECL_SIGWINCH
+	signal(SIGWINCH, handle_sigwinch);
+#endif
+}
+
+void tty_draw(void)
+{
+	size_t	i, j;
+	size_t	xscroll = 0;
+
+	tty_putp(cursor_invisible, 0);
+	tty_putp(cursor_home, 0);	/* move cursor home to avoid screen clutter */
+	tty_putp(carriage_return, 1);	/* move cursor to first column */
+	if (cursor_position >= xscroll + tty_columns)
+		xscroll = cursor_position - tty_columns + 1;
+	if (cursor_position < xscroll)
+		xscroll = cursor_position;
+	print_line(&query[xscroll], query_length - xscroll, 0, -1, -1);
+
+	if (selection - yscroll >= choices_lines)
+		yscroll = selection - choices_lines + 1;
```

Comment:
This is off by one from the code that was removed from the `PAGE_DOWN` and `END` branches in `selected_choice()`. Is this because it is now done prior to printing the choices?